### PR TITLE
Add tracking of block-and-text transitions.

### PR DESCRIPTION
**Pull request by ashercodeorg**
LOCAL TESTING:

image

### CODE REVIEW

**Comment by ashercodeorg** [comment]
TODO(@pcardune): Add an event fire on page load. Do similar for Game Lab. Also add an event fire on "Run". Also (ideally) populate the `user_id`.

**Comment by ashercodeorg** [comment]
Please confirm I'm injecting the event at the right place. Not being familiar with our JS, I found a place that worked, not necessarily the right place.

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -394,6 +395,11 @@ StudioApp.prototype.init = function (config) {
         var nonDropletError = false;
         // are we trying to toggle from blocks to text (or the opposite)
         var fromBlocks = this.editor.currentlyUsingBlocks;
+        var event = fromBlocks ? "block-to-text" : "text-to-block";
+        firehoseClient.putRecord(
+          'analysis-events',
+          {study: 'block-and-text-transitions', event: event}
+        );
```
Comment (Bjvanminnen) [inline apps/src/StudioApp.js]: This seems like the right place.

It does seem a little weird to me that we log this event using this `firehoseClient`, and then if the transition fails, a few lines later we `logToCloud` the error (which puts it in New Relic). Do we have guidelines somewhere on how to decide what the right place to log to is? Should we be moving some of the stuff that we log to NR to use this new logger?

Comment (ashercodeorg) [reply apps/src/StudioApp.js]: Though it does seem weird, I don't think anything needs to necessarily be changed. It seems appropriate to me to store data for analysis in Firehose and data on errors in New Relic.

Comment (Bjvanminnen) [reply apps/src/StudioApp.js]: I think part of why this feels particularly weird but we're storing two different parts of the same-ish thing in two places. FWIW, I agree that nothing necessarily needs to be changed here. However, I am still curious how one decides whether new logging belongs in Firehose or NR?

**Pull request merged**